specification/sbi_atee.adoc

```diff
@@ -13,6 +44,936 @@ running a VMM).
+TSM must be be in TSM_READY in order to process further ECALLs.
+
+A list of possible TSM states and the associated semantics appears below.
+
+[#table_tsm_states]
+.TSM States
+[%header,%autowidth]
+|===
+| TSM State          | Meaning
+
+| TSM_NOT_LOADED     | TSM has not been loaded on this platform.
+| TSM_LOADED         | TSM has been loaded, but has not yet been initialized.
+| TSM_READY          | TSM has been loaded & initialized, and is ready to accept ECALLs.
+|===
+
+
+=== Conversion of non-confidential memory to confidential memory
+
+Platform memory is non-confidential by default, and must be converted to confidential memory
+before use with TVMs. The conversion process is initiated by designating the host physical
+pages that to be converted, and then issuing fence operations to ensure that all outstanding
+TLB entries to the non-confidential memory are flushed across all VCPUs on the platform. This
+ensures that there's no overlapping mapping between the confidential and non-confidential memory
+regions on the platform.
+
+This requires the host to make three separate ECALLs to the TSM:
+
+. sbi_tee_tsm_convert_pages()
+. sbi_tee_tsm_global_fence()
+. sbi_tee_tsm_local_fence()
+
+The memory conversion process is complete when sbi_tee_tsm_local_fence() is successfully completed
+on the every VCPU on the platform.
+
+Converted memory can be assigned to TVMs, but cannot be repurposed for non-confidential operations
+unless it's reclaimed.
+
+=== TVM creation
+
+TVMs are created using the sbi_tee_create_tvm(). This creates a TVM, and sets with state "initializing".
+The host must assign confidential memory for page tables, payload mapping, and VCPUs before it can be
+transitioned into a "runnable" state.
+
+=== TVM memory management
+
+The host is responsible for the following:
+
+. Dontations for the page-table pool
+.. The host must ensure that the TSM has sufficient confidential memory for mapping and
+   managing TVM page-tables for the code and data payloads (sbi_tee_tvm_add_page_table_pages())
+. Defining memory regions for confidential, MMIO, and shared memory regions
+.. The host must declare the TVM physical address ranges for mapping of confidential and
+   non-confidential memory. There can be multiples ranges of each type, but no two regions
+   can overlap. Both confidential and non-confidential regions can be sparsely mapped; however,
+   any sparsely mapped confidential page that's demand-page can only be a demand-zero page. 
+   The `sbi_tee_tvm_memory_region()` can be used for this purpose.
+. Mapping of code and data payload to the TVM
+.. The host must copy the TVM code and data payload from non-confidential memory to confidential
+   memory by calling `sbi_tee_tvm_add_measured_pages()`
+. Mapping of zero-page confidential pages to the TVM
+.. The host can map zero-pages in a TVM confidential memory range at any point of time by
+   calling `sbi_tee_tvm_add_zero_pages()`.
```

Comment:
should add - the confidential memory page should not already be mapped, i.e. demand zero confidential pages may only be mapped for previously un-mapped pages.